set.mm

```diff
@@ -459446,8 +459446,7 @@ and the expression ( x e. A /\ x e. B ) ` .
      LALR parsers would normally detect them as shift-reduce or
      reduce-reduce conflicts, but the current implementation of metamath-knife
      cannot, and therefore requires those hints. $)
-  $( $j garden_path ( x   =>   ( ph ;
-        garden_path ( A   =>   ( ph ;
+  $( $j garden_path ( A   =>   ( ph ;
```

Comment:
I'm not sure I understand, but doesn't this contradict the rule https://us.metamath.org/mpeuni/co.html ?

Reply:
It's not a promise that we will take the path, it's saying that we might need to do a reduction at that point and yet still change our mind later. Without this directive (IIUC) it will stop at the first "reduce", meaning that anything which requires more to match can paint us into a corner. I don't think these directives can be used to make an incorrect automaton, but it causes the generator to do more work, and the resulting automaton can match more expressions. (I have seen garden_path directives cause construction to fail though, with an ambiguity error.)